### ISSUE TITLE
Questions about certificate generation/parsing

### ISSUE THREAD

**Issue by LuyaoZhong** [openssl/openssl#18800]
I'm programming certificate generation/parsing with C++. I'm suffering for looking for proper functions from a bunch of APIs, so I'd like to consult you experts.

1. extract extensions from certificate
Is there any funcions help extract all extensions from certificate and format it to a string like
```
[ v3_req ]
# Extensions to add to a certificate request 
basicConstraints = CA:FALSE 
keyUsage = nonRepudiation, digitalSignature, keyEncipherment 
subjectAltName = @alt_names 

[ alt_names ] 
DNS.1 = www.google.com 
```

2. add extensions to certificate
Is that possible to load above string and construct corresponding extensions and add them to cert?
I know these two functions is used to add extension, 
```cpp
X509V3_EXT_nconf_nid(...);  // define an extension
X509_add_ext(...);  // add an extension to cert
```
but I want to find a way like the command line does
```shell
$ openssl x509 -req -extensions v3_req -days 365 -CA  root-ca.pem -CAkey root-ca.key -set_serial 0 -in google.com.csr -out google.com.pem -extfile google.cnf
$ cat google.cnf 
[ v3_req ]
# Extensions to add to a certificate request 
basicConstraints = CA:FALSE 
keyUsage = nonRepudiation, digitalSignature, keyEncipherment 
subjectAltName = @alt_names 

[ alt_names ] 
DNS.1 = www.google.com  
#DNS.2 = *.baidu.com
```

4. convert RFC2253 name to X509_NAME
I only found a function which can convert X509_NAME to RFC2253, is there any function to do revert thing?
```cpp
X509_NAME_print_ex(buf.get(), name, 0 /* indent */, XN_FLAG_RFC2253);
```
Do you have any code snippet or anything related that I can refer to? I'll appreciate your help.

**Comment by ljluestc** [comment]
```

#include <openssl/x509.h>
#include <openssl/x509v3.h>

std::string extractExtensions(X509 *cert) {
    std::string extensionsStr;

    STACK_OF(X509_EXTENSION) *exts = X509_get0_extensions(cert);

    if (exts) {
        for (int i = 0; i < sk_X509_EXTENSION_num(exts); ++i) {
            X509_EXTENSION *ext = sk_X509_EXTENSION_value(exts, i);
            char *extStr = X509V3_EXT_print(NULL, ext, 0, 0);
            extensionsStr += extStr;
            OPENSSL_free(extStr);
        }
    }

    return extensionsStr;
}

bool addExtensionsToCert(X509 *cert, const std::string &extensionsStr) {
    // Parse the extensionsStr and create X509_EXTENSION objects
    // Add each extension to the certificate using X509_add_ext()

    return true; // Return true on success, false on failure
}

X509_NAME* convertRFC2253ToX509Name(const std::string &rfc2253Name) {
    X509_NAME *name = X509_NAME_new();
    if (!name) {
        // Handle error
        return nullptr;
    }

    if (X509_NAME_oneline(rfc2253Name.c_str(), NULL, 0) == NULL) {
        // Handle error
        X509_NAME_free(name);
        return nullptr;
    }

    return name;
}
```

**Comment by nhorman** [comment]
marking as inactive, will be closed at the end of the 3.4 dev cycle barring further input
